contracts/VaultV3.vy

```diff
@@ -640,6 +639,25 @@ def _assess_share_of_unrealised_losses(strategy: address, assets_needed: uint256
 
     return losses_user_share
 
+@internal
+def _withdraw_from_strategy(strategy: address, assets_to_withdraw: uint256):
+    """
+    This takes the amount denominated in asset and performs a {redeem}
+    with the corresponding amount of shares.
+
+    We use {redeem} because the Tokenized Strategies cannot natively take on
+    losses during {withdraw} without aditional non-4626 standard parameters.
+    """
+    # Need to get shares since we use redeem to be able to take on losses.
+    shares_to_redeem: uint256 = min(
+        # Use previewWithdraw since it should round up.
+        IStrategy(strategy).previewWithdraw(assets_to_withdraw), 
+        # And check against our actual balance.
+        IStrategy(strategy).balanceOf(self)
```

Comment:
should we use maxRedeem instead of balanceOf() even though maxWithdraw is always checked first

Reply:
https://github.com/yearn/yearn-vaults-v3/pull/175/commits/551e55496f9152a678a91900b1bc53ce6bb12b55